WEB/Src/DependencyCollector/DependencyCollector/Implementation/AzureSdk/AzureSdkDiagnosticsEventHandler.cs

```diff
@@ -102,8 +104,11 @@ public override void OnEvent(KeyValuePair<string, object> evnt, DiagnosticListen
                                 dependency.SetOperationDetail(evnt.Value.GetType().FullName, evnt.Value);
                             }
                         }
-                        else if (dependency.Type == CosmosDBResourceProviderNs)
+                        else if (dependency.Type == ClientCosmosDbDependencyType || dependency.Type == InternalCosmosDbDependencyType)
```

Comment:
shouldn't we have condition on _CosmosDBResourceProviderNs_. Something like...if resourcetype is DocumentDB then always set cosmosdb properties..no matter what is dependency type.

Reply:
why? we only have those two possible and they represent a combination of kind and az.namespace (CosmosDBResourceProviderNs). So currently it covers all possible cases and if we'd ever have more cases, we'll need to change it as well. We have tests that check it.

Reply:
`Kind` is not deciding factor here...we are not setting anything specific to a `kind`. That's why I think it makes sense to have this condition on the basis of `az.namespace` only.

Reply:
cosmos does not depend on kind, but code is not specific to cosmos. `depednency.Type` here is calculated based on `az.namespace`. If you believe this code has some problem, I'm happy to fix it. Otherwise, I'd prefer to keep it like this and follow the same patterns we have for EventHubs or ServiceBus.

Reply:
As discussed offline, It is intentional to keep this very restrictive condition to allow specific "kind"s  to set comsodb properties.
 In future, if we want to introduce a new type then appinsight sdk needs to be changed in order to support this.